diff -ur gmpc-0.18.96-o/data/gmpc.desktop.in gmpc-0.18.96/data/gmpc.desktop.in
--- gmpc-0.18.96-o/data/gmpc.desktop.in	2009-08-08 04:08:53.000000000 -0600
+++ gmpc-0.18.96/data/gmpc.desktop.in	2009-08-08 16:45:46.286212563 -0600
@@ -5,5 +5,5 @@
 Terminal=false
 Type=Application
 Icon=gmpc
-Categories=GNOME;AudioVideo;
+Categories=GTK;GNOME;AudioVideo;Audio;Player;
 StartupNotify=true
Only in gmpc-0.18.96/data: gmpc.desktop.in~
Only in gmpc-0.18.96/data: gmpc.desktop.in.orig
Only in gmpc-0.18.96/data: gmpc.desktop.in.rej
Only in gmpc-0.18.96/data: gmpc.desktop.in.rej~
Only in gmpc-0.18.96: gmpc-0.14.0.new
