diff -Nur gmpc-0.15.5.0.orig/configure.ac gmpc-0.15.5.0.locale/configure.ac
--- gmpc-0.15.5.0.orig/configure.ac	2007-11-29 06:52:03.000000000 -0700
+++ gmpc-0.15.5.0.locale/configure.ac	2008-04-26 18:04:16.000000000 -0600
@@ -147,7 +147,7 @@
 
 # i18n support
 dnl please keep them in alphabetical order
-ALL_LINGUAS="de fr gl it nl nb_NO pl ru"
+ALL_LINGUAS="de fr gl it nl nb pl ru"
 
 GETTEXT_PACKAGE=gmpc
 AC_SUBST(GETTEXT_PACKAGE)
