--- gmpc-0.14.0/src/Makefile.am.wig	2007-02-19 21:55:18.000000000 +0100
+++ gmpc-0.14.0/src/Makefile.am	2007-02-19 21:56:10.000000000 +0100
@@ -52,6 +52,7 @@
 INCLUDES= @EXTRA_CFLAGS@ -Wall\
 	-DPIXMAP_PATH=\""$(pkgdatadir)"\" \
 	-DGLADE_PATH=\""$(pkgdatadir)"\" \
+	-DPLUGINS_PATH=\""$(pkglibdir)"\" \
 	-DVERSION=\"@VERSION@\"
 
 LIBS= \
--- gmpc-0.14.0/src/main.c.wig	2007-02-19 21:53:40.000000000 +0100
+++ gmpc-0.14.0/src/main.c	2007-02-19 21:55:03.000000000 +0100
@@ -515,9 +515,7 @@
      *  load dynamic plugins 
      */
     /** Load the global installed plugins */
-    url = g_strdup_printf("%s/%s",GLADE_PATH, "plugins");
-    plugin_load_dir(url);
-    q_free(url);
+    plugin_load_dir(PLUGINS_PATH);
     /* user space dynamic plugins */
     url = g_strdup_printf("%s/.gmpc/plugins/",g_get_home_dir());
     /**
