diff -ur gmpc-0.15.0-o/src/main.c gmpc-0.15.0/src/main.c
--- gmpc-0.15.0-o/src/main.c	2007-05-19 14:40:09.000000000 -0600
+++ gmpc-0.15.0/src/main.c	2007-05-19 20:09:27.000000000 -0600
@@ -528,7 +528,7 @@
 
 #else
 	/** Load the global installed plugins */
-	url = g_strdup_printf("%s%c%s",GLADE_PATH,G_DIR_SEPARATOR, "plugins");
+	url = g_strdup(PLUGINS_PATH);
 #endif
 	plugin_load_dir(url);
 	q_free(url);
Only in gmpc-0.15.0/src: main.c~
Only in gmpc-0.15.0/src: main.c.orig
Only in gmpc-0.15.0/src: main.c.rej
Only in gmpc-0.15.0/src: main.c.rej~
diff -ur gmpc-0.15.0-o/src/Makefile.am gmpc-0.15.0/src/Makefile.am
--- gmpc-0.15.0-o/src/Makefile.am	2007-05-19 14:40:13.000000000 -0600
+++ gmpc-0.15.0/src/Makefile.am	2007-05-19 20:06:31.000000000 -0600
@@ -61,6 +61,7 @@
 INCLUDES= @EXTRA_CFLAGS@ -Wall\
 	-DPIXMAP_PATH=\""$(pkgdatadir)"\" \
 	-DGLADE_PATH=\""$(pkgdatadir)"\" \
+	-DPLUGINS_PATH=\""$(pkglibdir)"\" \
 	-DVERSION=\"@VERSION@\"
 
 LIBS= \
Only in gmpc-0.15.0/src: Makefile.am.orig
