diff -urN gmpc-0.15.5.0/src/main.c gmpc-0.15.5.0.new/src/main.c
--- gmpc-0.15.5.0/src/main.c	2007-11-29 14:52:03.000000000 +0100
+++ gmpc-0.15.5.0.new/src/main.c	2008-02-04 09:37:59.435020318 +0100
@@ -546,7 +546,7 @@
 
 #else
 	/** Load the global installed plugins */
-	url = g_build_path(G_DIR_SEPARATOR_S, GLADE_PATH,"plugins",NULL);
+	url = g_strdup(PLUGINS_PATH);
 #endif
     if(load_plugins)
         plugin_load_dir(url);
diff -urN gmpc-0.15.5.0/src/Makefile.am gmpc-0.15.5.0.new/src/Makefile.am
--- gmpc-0.15.5.0/src/Makefile.am	2007-11-29 14:52:03.000000000 +0100
+++ gmpc-0.15.5.0.new/src/Makefile.am	2008-02-04 09:38:44.441154350 +0100
@@ -81,6 +81,7 @@
 INCLUDES= @EXTRA_CFLAGS@ -Wall\
 	-DPIXMAP_PATH=\""$(pkgdatadir)"\" \
 	-DGLADE_PATH=\""$(pkgdatadir)"\" \
+	-DPLUGINS_PATH=\""$(pkglibdir)"\" \
 	-DVERSION=\"@VERSION@\"\
 	-DGMPC_BUILD=1
 
