--- gmpc-11.8.16/src/./playlist3.c~	2011-08-17 06:09:55.000000000 +0300
+++ gmpc-11.8.16/src/./playlist3.c	2012-01-28 20:22:19.595024784 +0200
@@ -814,8 +814,7 @@
             if (id)
             {
                 string =
-                    g_strdup_printf("[%s] %s - %s %s", gmpc_profiles_get_name(gmpc_profiles, id), _("GMPC"),
-                    _("Connected to"), mpd_get_hostname(mi));
+                    g_strdup_printf("%s - %s %s [%s]", _("GMPC"), _("Connected to"), mpd_get_hostname(mi), gmpc_profiles_get_name(gmpc_profiles, id));
                 g_free(id);
             }
         }
@@ -833,8 +832,7 @@
             if (id)
             {
                 string =
-                    g_strdup_printf("[%s] %s - %s", gmpc_profiles_get_name(gmpc_profiles, id), _("GMPC"),
-                    _("Disconnected"));
+                    g_strdup_printf("%s - %s [%s]", _("GMPC"), _("Disconnected"), gmpc_profiles_get_name(gmpc_profiles, id));
                 g_free(id);
             }
         }
